src/flags.rs

```diff
@@ -174,26 +174,106 @@ impl<'this> RustcCodegenFlags<'this> {
-                if self.no_vectorize_slp {
-                    push_if_supported("-fno-slp-vectorize".into());
+            }
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fPIC
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fPIE
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-mdynamic-no-pic
+            if let Some(value) = self.relocation_model {
+                let cc_flag = match value {
+                    "pic" => Some("-fPIC"),
+                    "pie" => Some("-fPIE"),
+                    "dynamic-no-pic" => Some("-mdynamic-no-pic"),
+                    _ => None,
+                };
+                if let Some(cc_flag) = cc_flag {
+                    push_if_supported(cc_flag.into());
                 }
+            }
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fembed-bitcode
+            if let Some(value) = self.embed_bitcode {
+                let cc_val = if value { "all" } else { "off" };
+                push_if_supported(format!("-fembed-bitcode={cc_val}").into());
+            }
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fno-omit-frame-pointer
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-fomit-frame-pointer
+            if let Some(value) = self.force_frame_pointers {
+                let cc_flag = if value {
+                    "-fno-omit-frame-pointer"
+                } else {
+                    "-fomit-frame-pointer"
+                };
+                push_if_supported(cc_flag.into());
+            }
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-dead_strip
+            if let Some(false) = self.link_dead_code {
+                push_if_supported("-dead_strip".into());
+            }
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-mno-red-zone
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-mred-zone
+            if let Some(value) = self.no_redzone {
+                let cc_flag = if value { "-mno-red-zone" } else { "-mred-zone" };
+                push_if_supported(cc_flag.into());
+            }
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-msoft-float
+            // https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-mno-soft-float
+            if let Some(value) = self.soft_float {
+                let cc_flag = if value {
+                    "-msoft-float"
+                } else {
+                    "-mno-soft-float"
+                };
+                push_if_supported(cc_flag.into());
+            }
+        }
+
+        // Compiler-exclusive flags
+        match family {
+            ToolFamily::Clang { .. } => {
```

Comment:
Please document here that while GCC supports the `-fprofile-generate=` and `-fprofile-use=` flags, we don't use them because they'll likely be incompatible with the Rust flag (while both Clang and Rust use LLVM under the hood, so they'll likely be compatible).

Reply:
Ah I did think about that, sure I'll add it in

Reply:
there you go